### ISSUE TITLE
Replace `aesara.tensor.sub` by `aesara.tensor.subtract`

### ISSUE THREAD

**Issue by rlouf** [aesara-devs/aesara#1423]
Add `sub` as an alias and add a deprecation warning.

https://numpy.org/doc/stable/reference/generated/numpy.subtract.html#numpy-subtract

**Comment by AtharvaMaskar26** [comment]
Hey!
I would like to work on this issue, is it still open?

**Comment by brandonwillard** [comment]
> Hey! I would like to work on this issue, is it still open?

Yes, feel free.  Just know that most of our team is currently focusing on our work on https://github.com/outlines-dev/outlines.

**Comment by AtharvaMaskar26** [comment]
Hey!
I am getting the following error when I'm trying to commit to my forked repository:

`aesara\link\c\cmodule.py:2459: error: Unused "type: ignore" comment`

![image](https://github.com/aesara-devs/aesara/assets/107984209/4757f9e0-f209-48b8-a6e5-61642614f3a4)

Can someone help me with it?

**Comment by brandonwillard** [comment]
Sometimes that error appears due to the somewhat irregular way that `pre-commit` uses `mypy` (e.g. on only the files that have changed).  You can ignore/disable the check in this instance and see if the error appears in CI after submitting a PR.

Also, note that there is an open PR associated with this, so you can always work from that.

Otherwise, sorry for the late reply; we're running another very active project (i.e. https://outlines-dev.github.io/outlines/) at the moment, and that is our primary focus.
